### PR TITLE
[KeyVault] Redesign HSM cmdlets

### PR DESCRIPTION
**Pull request by isra-fel**
## Description

According to our discussion with KeyVault team, here's the summary of the changes:
(I organized code changes in different commits. I suggest review them one by one.)

1. Management plane
    - Rename, add "KeyVault" prefix
    - Old: `New-AzManagedHsm`
    - New: `New-AzKeyVaultManagedHsm`
1. Key operations
    - Merge with existing "KeyVaultKey" cmdlets
    - Make -VaultName and -HsmName exclusive for different scenarios
    - Old: `Add-AzKeyVaultKey` and `Add-AzManagedHsmKey`
    - New: Merge both into `Add-AzKeyVaultKey`
1. RBAC
    - Rename, use "KeyVault" as prefix
    - Old: `Get-AzManagedHsmRoleDefinition`
    - New: `Get-AzKeyVaultRoleDefinition`
1. Full backup & restore
    - Rename, use "KeyVault" as prefix
    - Old: `Backup-AzManagedHsm`
    - New: `Backup-AzKeyVault`
1. Security domain
    - Rename, use "KeyVault" as prefix
    - Use verb Export and Import
    - Old: `Backup-AzManagedHsmSecurityDomain` and `Restore-AzManagedHsmSecurityDomain`
    - New: `Export-AzKeyVaultSecurityDomain` and `Import-AzKeyVaultSecurityDomain`

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/KeyVault/KeyVault/Commands/AddAzureKeyVaultKey.cs [hunk 1]
```diff
@@ -198,7 +249,7 @@ public class AddAzureKeyVaultKey : KeyVaultCmdletBase
         /// Key tags
         /// </summary>
         [Parameter(Mandatory = false,
-           HelpMessage = "A hashtable representing key tags.")]
+            HelpMessage = "A hashtable representing key tags.")]
```
Comment (msJinLei) [inline src/KeyVault/KeyVault/Commands/AddAzureKeyVaultKey.cs:252]: redundant space

File: src/KeyVault/KeyVault/Commands/UpdateAzureManagedHsm.cs [hunk 2]
```diff
@@ -68,7 +68,7 @@ public override void ExecuteCmdlet()
 
             if (existingResource == null)
             {
-                throw new Exception(string.Format("A managed HSM with name '{0}' in resource group '{1}' does not exist. Please use New-AzManagedHsm to create a managed HSM with these properties.", this.Name, this.ResourceGroupName));
+                throw new Exception(string.Format("A managed HSM with name '{0}' in resource group '{1}' does not exist. Please use New-AzKeyVaultManagedHsm to create a managed HSM with these properties.", this.Name, this.ResourceGroupName));
```
Comment (msJinLei) [inline src/KeyVault/KeyVault/Commands/UpdateAzureManagedHsm.cs:71]: It better to use resource string

File: src/KeyVault/KeyVault/Commands/BackupAzureKeyVaultKey.cs [hunk 3]
```diff
@@ -43,69 +44,103 @@ public class BackupAzureKeyVaultKey : KeyVaultCmdletBase
...
+            {
+                BackupKeyVaultKey();
+            }
+            else
+            {
+                BackupHsmKey();
+            }
+        }
+
+        private void NormalizeParameterSets()
```
Comment (msJinLei) [inline src/KeyVault/KeyVault/Commands/BackupAzureKeyVaultKey.cs:125]: You can put NormalizeParameterSets into base class

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/KeyVault/KeyVault/Commands/BackupAzureKeyVaultKey.cs (hunk 3 above)
Comment (isra-fel) [reply src/KeyVault/KeyVault/Commands/BackupAzureKeyVaultKey.cs:125]: I wish I could, but the logic of this method is slightly different in every cmdlet.
For example, some cmdlets have `-ResourceId` and some don't, some even have `-HsmResourceId` as well.
I'm afarid we won't get much benifit to write it in base class.

File: src/KeyVault/KeyVault/Commands/UpdateAzureManagedHsm.cs (hunk 2 above)
Comment (isra-fel) [reply src/KeyVault/KeyVault/Commands/UpdateAzureManagedHsm.cs:71]: Cool. I'll fix it.

**Review by msJinLei — APPROVED** [review]

**Pull request merged**